### PR TITLE
SONARIAC-534 Rule S2612: Dangerous chmod option on COPY or ADD instructions

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

**Comment by rudy-regazzoni-sonarsource** [comment]
And as usual, a better name to be found for the class ^^' `PermissionCheck` is too generic, maybe `ChmodPermissionCheck` but then it feel too specific.

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/checks/AddAndCopyPermissionCheckTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  @Test
+  void test_add() {
+    DockerVerifier.verify("AddAndCopyPermissionCheckTest/Dockerfile_add", new AddAndCopyPermissionCheck());
+  }
+
+  @Test
+  void test_copy() {
+    DockerVerifier.verify("AddAndCopyPermissionCheckTest/Dockerfile_copy", new AddAndCopyPermissionCheck());
+  }
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/test/java/org/sonar/iac/docker/checks/AddAndCopyPermissionCheckTest.java:34]: I'm not sure if a dedicated test is nesseary. This only increases the maintenance effort. I would keep only one test and show, with 1-2 lines if `COPY` that there is no difference between `ADD` and `COPY`.

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM! Please see comments about naming and the superfluous test file.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/AddAndCopyPermissionCheck.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.docker.tree.api.AddInstruction;
+import org.sonar.iac.docker.tree.api.CopyInstruction;
+import org.sonar.iac.docker.tree.api.TransferInstruction;
+import org.sonar.iac.docker.utils.ArgumentUtils;
+
+@Rule(key = "S2612")
+public class AddAndCopyPermissionCheck implements IacCheck {
```
[the remark this answers is not in the document]
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/AddAndCopyPermissionCheck.java:34]: Good point, I forgot to check in other analyzers, thanks for reminding !

**Pull request merged**